app_dart/lib/src/request_handlers/get_branches.dart

```diff
@@ -39,30 +41,43 @@ import '../service/datastore.dart';
 class GetBranches extends RequestHandler<Body> {
   GetBranches({
     required super.config,
+    required this.branchService,
     this.datastoreProvider = DatastoreService.defaultProvider,
     this.processRunner,
   });
 
+  final BranchService branchService;
   final DatastoreServiceProvider datastoreProvider;
   ProcessRunner? processRunner;
 
   static const Duration kActiveBranchActivity = Duration(days: 60);
 
+  bool isRecent(Branch b) {
+    return DateTime.now().millisecondsSinceEpoch - b.lastActivity! < kActiveBranchActivity.inMilliseconds ||
```

Comment:
Instead of hard coding branches, you can do

```dart
Config.defaultBranch(slug) == b.name
```

Reply:
Umm my understanding is that the slug needs to be first identified as one of `cocoonSlug`, `engineSlug` etc. to be able to retrieve a default branch.
Umm I thought here we are filtering from branches across all repos (noted in the initial design in the comments section), and therefore a specific slug of one repo might not be applicable.

Reply:
If that's the case, we should a check for `Config.supportedRepos` to prevent unnecessary processing

Reply:
umm changed to `config.supportedRepos.map((repo) => Config.defaultBranch(repo)).toSet().contains(b.name)` to comply with the requirements.

Umm I thought using main and master would be easier but maybe we support default branches other than main and master?

Reply:
synced with Casey, we shouldn't use the word master. And could potentially pass the repo as one of the parameters.